zipline/utils/events.py

```diff
@@ -594,13 +605,14 @@ def get_nth_to_last_trading_day_of_month(self, dt, env):
             # We already computed the last day for this month.
             return self.day
 
-        if not self.td_delta:
-            self.day = self.get_last_trading_day_of_month(dt, env)
-        else:
-            self.day = env.add_trading_days(
-                self.td_delta,
-                self.get_last_trading_day_of_month(dt, env),
-            ).date()
+        self.day = self.get_last_trading_day_of_month(dt, env)
+        if self.td_delta and self.day:
```

Comment:
Is it worth is to make a base class off which to subclass these two rules (like it is done with the week rule)? Or should it just share the method defined somewhere